tests/test_freqtradebot.py

```diff
@@ -4632,6 +4641,346 @@ def make_sure_its_651(*args, **kwargs):
+    assert len(trades) == 1
+    # Assert trade is as expected (averaged dca)
+
+    trade = Trade.query.first()
+    assert trade
+    assert trade.open_order_id is None
+    assert trade.amount == 50
+    assert trade.open_rate == 11
+    assert trade.stake_amount == 550
+    assert pytest.approx(trade.realized_profit) == -152.375
+    assert pytest.approx(trade.close_profit_abs) == -152.375
+
+    orders = Order.query.all()
+    assert orders
+    assert len(orders) == 2
+    # Make sure the closed order is found as the second order.
+    order = trade.select_order('sell', False)
+    assert order.order_id == '601'
+
+    amount = 50
+    ask = 16
+    closed_sell_dca_order_2 = {
+        'ft_pair': pair,
+        'status': 'closed',
+        'ft_order_side': 'sell',
+        'side': 'sell',
+        'type': 'limit',
+        'price': ask,
+        'average': ask,
+        'amount': amount,
+        'filled': amount,
+        'cost': amount * ask,
+        'ft_is_open': False,
+        'id': '602',
+        'order_id': '602'
+    }
+    mocker.patch('freqtrade.exchange.Exchange.create_order',
+                 MagicMock(return_value=closed_sell_dca_order_2))
+    mocker.patch('freqtrade.exchange.Exchange.fetch_order',
+                 MagicMock(return_value=closed_sell_dca_order_2))
+    mocker.patch('freqtrade.exchange.Exchange.fetch_order_or_stoploss_order',
+                 MagicMock(return_value=closed_sell_dca_order_2))
+    assert freqtrade.execute_trade_exit(trade=trade, limit=ask,
+                                        sell_reason=SellCheckTuple(sell_type=SellType.STOP_LOSS),
+                                        sub_trade_amt=amount)
+    # Assert trade is as expected (averaged dca)
+
+    trade = Trade.query.first()
+    assert trade
+    assert trade.open_order_id is None
+    assert trade.amount == 50
+    assert trade.open_rate == 11
+    assert trade.stake_amount == 550
+    assert pytest.approx(trade.realized_profit) == -152.375
+    assert pytest.approx(trade.close_profit_abs) == 94.25
+    orders = Order.query.all()
+    assert orders
+    assert len(orders) == 3
+
+    # Make sure the closed order is found as the second order.
+    order = trade.select_order('sell', False)
+    assert order.order_id == '602'
+    assert trade.is_open is False
+
+
+@pytest.mark.parametrize('orders, res', [
+    (
+        (
+            # side ampunt, price
+            ('buy', 100, 10),
+            ('buy', 100, 15),
+            ('sell', 50, 12),
+            ('sell', 100, 20),
+            ('sell', 50, 5),
+        ),
+        (
+            # amount, open_rate, stake_amount, cumulative_profit, realized_profit
+            (100.0, 10.0, 1000.0, 0.0, None,),
+            (200.0, 12.5, 2500.0, 0.0, None,),
+            (150.0, 12.5, 1875.0, -28.0625, -28.0625,),
+            (50.0, 12.5, 625.0, 713.8125, 741.875,),
+            (50.0, 12.5, 625.0, 713.8125, 336.625,),
+        )
```

Comment:
From an test org point of view - i think i would combine these two arrays into one long array - where the first 3 columns are "order" - and the remainder is price.

That will make it a lot easier for the "reader" to immediately identify after which step we expect which result - while also removing the need to `res[idx]` all asserts.

Reply:
Planning to do different modes of Partial sells. So that time, input can be reused. used zip.